alfalfa_worker/Dockerfile

```diff
@@ -170,50 +177,39 @@ RUN git clone --branch $ASSIMULO_TAG --depth 1 https://github.com/modelon-commun
 # Install PyFMI
 RUN git clone --branch $PYFMI_TAG --depth 1 https://github.com/modelon-community/PyFMI.git \
     && cd PyFMI \
-    && python setup.py install \
+    && poetry run python setup.py install \
     && cd .. && rm -rf PyFMI
 
-
-# There are some tweaks because Ubuntu 20.04 has a PEP404 error with the version.
-# See https://bugs.launchpad.net/ubuntu/+source/python-debian/+bug/1926870
-# The below installs poetry 1.1.15 because 1.2.1 has the error.
-# It then copies over the poetry files and installs the dependencies with --no-dev
-RUN pip3 install poetry==1.1.15
-
-# Copy and install poetry-based dependencies
-COPY pyproject.toml $BUILD_DIR
-COPY poetry.lock $BUILD_DIR
-
-RUN poetry config virtualenvs.create false \
-    && poetry install --no-dev
-
-# reinstall setuptools
-RUN pip3 install setuptools==65.3.0
-
 WORKDIR /alfalfa
+
 # Include the path to where alfalfa_worker is so that system calls can find it.
 ENV  PYTHONPATH="/alfalfa:${PYTHONPATH}"
 
 COPY . /alfalfa
 COPY ./deploy/wait-for-it.sh /usr/local/wait-for-it.sh
-COPY ./deploy/start_worker.sh start_worker.sh
 
+# update the path to python, which is in the Poetry virtual environment.
+# The path (mfknN3Ol) is generated by Poetry and should be updated if the
+# install path changes.
+#
+# The dependency warning still exists, but it is a warning. Eventually this
+# will be resolved when we move to a newer version of Ubuntu.
+# See https://bugs.launchpad.net/ubuntu/+source/python-debian/+bug/1926870
+# Error: RequestsDependencyWarning: urllib3 (1.26.13) or chardet (3.0.4) doesn't match a supported version!
+#        warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported
+ENV PATH=/home/alfalfa/.cache/pypoetry/virtualenvs/alfalfa-mfknN3Ol-py3.8/bin:$PATH
```

Comment:
this is deterministic?

Reply:
I questioned this myself. It is deterministic as long as we keep the path /alfalfa. Crazy, right?!

Reply:
huh

Reply:
Here is the poetry "generate_env_name" code for reference. hash created from current working directory.

https://github.com/python-poetry/poetry/blob/5f8b9da15dc13e44ac5bf3f2cd9ac1f72b3178ac/src/poetry/utils/env.py#L1147